### PR TITLE
Move DateField, DateTimeField, TimeField comment buttons to be right next to the fields. Fix #9543

### LINKED ISSUE #9543: Too much spacing for comment button on DateFields
### Issue Summary

The comment button is directly behind the input field in all input fields. But for DateFields there is a large space between the input field and the comment button.

[Peek 2022-10-29 15-29.webm](https://user-images.githubusercontent.com/4333770/198834227-40aec86f-3b57-4e20-ba18-b011f8d965df.webm)

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Create a page model with an `DateField`
    ```python
    class BlogPage(Page):
        date = models.DateField("Post date")
    ```
3. Migrate and start the server
4. Open den Wagtail admin interface
5. Enable the comment mode

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.10.8
-   Django version: 4.1.1
-   Wagtail version: 4.0.4
-   Browser version: Firefox 106 and Chrome  107

### PR DESCRIPTION
**Pull request by Theresa-o**
PR for issue https://github.com/wagtail/wagtail/issues/9543

ISSUE
There was a large space difference between the datefield input and the comment button

CAUSE OF ISSUE
All input fields had a `width: 100%` but the datefield wield was adjusted to `width: auto` so this caused the space between the datefield and the comment button as seen in the screenshot below

SOLUTION
Added the below code to change the position of the comment button

`.w-field--date_field,
.w-field--date_time_field,
.w-field--time_field {
  .w-field__comment-button {
    position: relative;
  }
}`

RESULT

BEFORE
wagcommentissuereproduced

AFTER
wagcommentissueresolved

### CODE REVIEW

File: client/scss/components/_button.scss
Comment (lb-) [inline client/scss/components/_button.scss:122]:
```suggestion
      width: 3em:
```

File: client/scss/components/_dropdown.legacy.scss
Comment (zerolab) [reply client/scss/components/_dropdown.legacy.scss:114]: note that these are remnants from #9500, and as such they do not belong here.

Suggestion: always start new work off the latest `main` branch

Comment (Theresa-o) [reply client/scss/components/_dropdown.legacy.scss:114]: @zerolab Sorry about this, I've been trying to delete the files

File: client/scss/components/_button.scss
Comment (lb-) [inline client/scss/components/_button.scss:122]:
```suggestion
      width: 3em;
```

File: client/scss/components/forms/_field-comment-control.scss
Comment (lb-) [inline client/scss/components/forms/_field-comment-control.scss:51]: Not sure if it's related but we have an override for these field types in `client/scss/components/forms/_input-text.scss`

https://github.com/wagtail/wagtail/blob/b9a5f9e333e6219141471f03d64aa16d927fc35b/client/scss/components/forms/_input-text.scss#L37-L44

Comment (Theresa-o) [reply client/scss/components/forms/_field-comment-control.scss:51]: Yes, this is related and this is why the width of the `datefield` input is smaller than the width of the other input fields as seen below;

wagcommentissuereproduced

Becuase of the adjustment of width here

 // Make non-text field types with custom widgets have a smaller width. 
 .w-field--date_field, 
 .w-field--date_time_field, 
 .w-field--time_field { 
   input { 
     width: auto; 
   } 
 } 

The comment button and `datefield` input has a noticeably large space in between because of the below code (client/scss/components/forms/_field-comment-control.scss):

.w-field__comment-button {
  $root: &;

  @include transition(opacity 0.2s ease);
  position: absolute;

**Comment by lb-** [comment]
As per https://github.com/wagtail/wagtail/issues/9543#issuecomment-1299118965 - we need a bit more info on that issue as to whether this is the right way to 'fix' this. It is possible that there is some higher level design nuance that the bug is missing.

My gut says we should be changing the chooser comment button but I want to wait until we get feedback on the original issue from some others in the core team before we do that.

**Comment by Theresa-o** [comment]
@lb- Thanks LB for reviewing, I await the design decision.

File: client/scss/components/forms/_field-comment-control.scss
Comment (thibaudcolas) [inline client/scss/components/forms/_field-comment-control.scss:56]: Looking great @Theresa-o, this is just missing the positioning tweaks a few lines above so the button sits where intended vertically.

**Review by thibaudcolas — APPROVED** [review]

**Comment by thibaudcolas** [comment]
Cherry-picked to `stable/4.1.x`

**Pull request merged**
